packages/search-ui/src/SearchDriver.js

```diff
@@ -151,7 +165,8 @@ export default class SearchDriver {
     // to the correct default values for the initial UI render
     this.state = {
       ...this.state,
-      ...searchParameters
+      ...searchParameters,
+      ...{ a11yNotificationMessages }
```

Comment:
actually - I'm not 100% sure, but should this really be in state? Or does it feel more like an action?

In theory you are calling functions, e.g. `a11yNotificationMessages.searchResults({ ... })` which makes me think this belongs more in `actions` than in state.

Reply:
Please, take the following with a grain of salt, I'm not sure if I'm right or not.

They don't look like state - they don't change after initialization.
They also don't look like actions - you don't call them directly to change state.
Aren't they more like a part of a config?

Reply:
Yeah, I think I actually agree with you. I don't know how else to pass it down to the `withSearch` HOC though - since state and action are what get passed down, and we need views/users to be able to access the message functions somehow.

Should we maybe pass down a whole new key like `getMessages` or `driver.messages`?... 😬

Reply:
> we need views/users to be able to access the message functions somehow
Why? I thought users would add these functions to config once and then not touch it anywhere in the components (in most cases). If (in rare cases) users need to change the a11y message for some reason manually, they would call one universal action a11yNotify.

Reply:
Yes, that was my impression too. I thought the only thing we were going to do with them is have the SearchDriver write them out to a DOM element.

Reply:
My impression was that we needed views to have access to the message functions so they can be called within views. I particularly need this not for the "search results" announcement, which can be done in SearchDriver.js but for the "more filters" announcement.

"More Filters" logic is handled entirely within `react-search-ui`'s Facet.js and is almost entirely a view-related concern. Folks rolling their own Vue or non-React components will not use our implementation and therefore the announcement doesn't belong in the top level `search-ui` package.

However, we still need it to support custom messages passed to `a11yNotificationMessages`, and therefore the component needs access to said obj of message functions - presumably through the HOC.

Does that make use case make sense to you both?

Reply:
@constancecchen It does, let me post an example of how I would handle this.

Reply:
This is my suggestion:

SearchDriver:
- Add an action called "a11yNotification" (or something similar) that accepts a string to be read in notifications. i.e., `allyNotification("More was clicked, showing whatever")`

Facet.js
https://github.com/elastic/search-ui/blob/master/packages/react-search-ui/src/containers/Facet.js#L48
- Call `allyNotification` with a message on `handleClickMore`.

How does that sound?

Reply:
If we felt we need to keep all messages together, `allyNotification` could take a string that corresponds to a key in `a11yNotificationMessages`, so instead of reading whatever is passed literally, it would look up a message.

Reply:
@JasonStoltz It does, and that's what I've done in https://github.com/elastic/search-ui/pull/355/commits/114921e07dbdc86521e0e605994f1fd9d1fce5e1 - the issue is that I still need access to `a11yNotificationMessages` (the message functions) which means passing it down via either state or actions.

The original impetus behind `a11yNotificationMessages` was to let devs customize the screen reader message (for localization, etc.). If I don't use it, I'm passing in a hard coded string and negating that customization.

Reply:
Hm - the only problem I see with the string/key lookup is that (also per the original [issue writeup](https://github.com/elastic/search-ui/issues/325)) - messages aren't just a static string, we're passing in arguments in order to compose a string, and some of these args are only available in the view itself.

Unless I'm missing some way of both looking up a function and passing in args at the same time? 😕

Reply:
@constancecchen I don't think you should / need to be working with `a11yNotificationMessages` in that component.

```
       const message = this.props.a11yNotificationMessages.moreFilters({
        optionsCount,
        showingAll
      });
      this.props.a11yNotify(message);
```

Could just be:

```
      this.props.a11yNotify("moreFilters", { optionsCount, showingAll});
```

Which is saying "read the moreFilter message, here are the corresponding options".

Reply:
Ahhh! 🤦‍♀ Yep, that would do it. I don't know why passing more than one param didn't occur to me sooner, thanks so much Jason!!

Reply:
This makes [the `a11yNotify` action feel much more robust](https://github.com/elastic/search-ui/pull/355/commits/78ffd85d6edc1efab82d9012a40bc5f4d777ab47) now too. Thanks a billion Jason! ✨

Reply:
Also, just one more follow-up to this refactor that I only just considered - if users want to roll their own completely custom a11y notification for, e.g. their completely custom component - they should accomplish this by extending the `a11yNotificationMessages` obj, correct? So, for example:

```js
SearchDriver({
  a11yNotificationMessages: {
    customMessage: () => 'Hello world',
  },
});
```

This means that:

1. We're now limiting the `a11yNotify` action to not simply allow strings - originally I was thinking users could do something like `a11yNotify('Whatever I want!')`, but that will no longer be the case with this architecture. Are we good with it? (I'm good with it personally, just checking that we're all on the same page.)

2. Per 78ffd85, we're now limiting `a11yNotificationMessages` to only contain functions. For example, if someone tried to pass `customMessage: 'Hello world'`, a type error would get thrown. Are we okay with this restriction? Or do we want to add a `typeof function` check to allow both primitive strings _and_ functions? 🤔 (Feels a bit overkill to me and I'm fine with the limitation, but also wanted to check here that we all agree.)

@JasonStoltz @yakhinvadim - would love your thoughts! 😀

Reply:
@constancecchen Probably clearer for users to just always use functions. Easier to document, less code, smaller API surface.

Reply:
(2) I agree with Jason

Reply:
(1) I'm not 100% sure about this, more like 70%, but if you both think that that's ok, I'm in.